src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryServiceSupport.scala

```diff
@@ -102,16 +98,14 @@ trait LibraryServiceSupport extends DataUseRestrictionSupport with LazyLogging {
 
   // wraps the ontologyDAO call, handles Nones/nulls, and returns a [Future[Seq].
   // the Seq is populated if the leaf node exists and has parents; Seq is empty otherwise.
-  def lookupParentNodes(leafId:String, ontologyDAO: OntologyDAO)(implicit ec: ExecutionContext):Future[Seq[TermParent]] = {
-    ontologyDAO.search(leafId) map {
-      case Some(terms) if terms.nonEmpty =>
+  def lookupParentNodes(leafId:String, ontologyDAO: OntologyDAO)(implicit ec: ExecutionContext):Seq[TermParent] = {
+    Try(ontologyDAO.search(leafId)) match {
+      case Success(terms) if terms.nonEmpty =>
         terms.head.parents.getOrElse(Seq.empty)
-      case None => Seq.empty[TermParent]
-    } recoverWith {
-      case ex:Exception => {
+      case Success(empty) => Seq.empty[TermParent]
+      case Failure(ex) =>
         logger.warn(s"exception getting term and parents from ontology: ${ex.getMessage}")
-        Future(Seq.empty[TermParent])
-      }
+        Seq.empty[TermParent]
```

Comment:
A better solution is to change the ontology index itself, such that it holds both the label and the id for any node's parents. Right now it only holds the id - so we have to make these extra queries to populate the labels.

Reply:
See https://github.com/broadinstitute/consent/pull/247